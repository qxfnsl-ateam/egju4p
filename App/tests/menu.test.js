import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Menu from '../app-src/js/menu';



describe('<Menu />', () => {
	it('Render empty Menu', () => {
		const list = [{value:1, title:'Ityem 1'}, {value:2, title:'Ityem 2'}, {value:3, title:'Ityem 3'}];
		const expanded = 2;
		const prefix = 'prefix-';
		const cmp = shallow(
			<Menu list={list} expanded={expanded}  prefix={prefix} onExpand={jest.fn()} onAction={jest.fn()} />
		);

		const eItems = cmp.find('li[value][onClick]');
		expect(eItems.length).toBe(list.length);
		eItems.forEach((eItem, i) => {
			expect(eItem.prop('value')).toBe(list[i].value);
			expect(eItem.key()).toMatch(prefix + list[i].value);
			expect(eItem.text()).toMatch(list[i].title);
			expect(eItem.hasClass('on')).toBe(false);
			expect(eItem.find('.staff-menu-empty').length).toBe(list[i].value == expanded ? 1 : 0);
		});

		const eButs = cmp.find('button[onClick].staff-menu-exp');
		expect(eButs.length).toBe(list.length);
		eButs.forEach((eBut, i) => {
			expect(eBut.hasClass('on')).toBe(list[i].value == expanded);
		});

		const eChilds = cmp.find('.staff-menu-empty');
		expect(eChilds.length).toBe(1);
	})


	it('Render Menu', () => {
		const list = [{value:1, title:'Ityem 1'}, {value:2, title:'Ityem 2'}, {value:3, title:'Ityem 3'}];
		const current = 1;
		const expanded = 2;
		const prefix = 'prefix-';
		const cmp = shallow(
			<Menu list={list} current={current} expanded={expanded} prefix={prefix} onExpand={jest.fn()} onAction={jest.fn()}>
				<embed />
			</Menu>
		);

		const eItems = cmp.find('li[value][onClick]');
		expect(eItems.length).toBe(list.length);
		eItems.forEach((eItem, i) => {
			expect(eItem.prop('value')).toBe(list[i].value);
			expect(eItem.key()).toMatch(prefix + list[i].value);
			expect(eItem.text()).toMatch(list[i].title);
			expect(eItem.hasClass('on')).toBe(list[i].value == current);
			expect(eItem.find('embed').length).toBe(list[i].value == expanded ? 1 : 0);
			//expect(eItem.contains(<embed />)).toBe(list[i].value == expanded);
		});

		const eButs = cmp.find('button[onClick].staff-menu-exp');
		expect(eButs.length).toBe(list.length);
		eButs.forEach((eBut, i) => {
			expect(eBut.hasClass('on')).toBe(list[i].value == expanded);
		});

		const eChilds = cmp.find('embed');
		expect(eChilds.length).toBe(1);
		//expect(cmp.contains(<embed />)).toBe(true);
	})


	it('Clickability', () => {
		const list = [{value:1, title:'Ityem 1'}, {value:2, title:'Ityem 2'}, {value:3, title:'Ityem 3'}];
		const current = 1;
		const expanded = 2;
		const prefix = 'prefix-';
		const onExpand = jest.fn();
		const onAction = jest.fn();
		const cmp = shallow(
			<Menu list={list} current={current} expanded={expanded} prefix={prefix} onExpand={onExpand} onAction={onAction}>
				<embed />
			</Menu>
		);

		const eItems = cmp.find('li[value][onClick]');
		expect(eItems.length).toBe(list.length);
		eItems.forEach((eItem, i) => {
			expect(eItem.prop('onClick')).toBeDefined();
			const evt = {target:{}};
			eItem.simulate('click', evt);
			expect(onAction).toBeCalledWith(evt);
		});

		const eButs = cmp.find('button[onClick].staff-menu-exp');
		expect(eButs.length).toBe(list.length);
		eButs.forEach((eBut, i) => {
			expect(eBut.prop('onClick')).toBeDefined();
			const evt = {target:{}};
			eBut.simulate('click', list[i].value == expanded, evt);
			expect(onExpand).toBeCalled();
		});
	})


	it('DOM tree', () => {
		const list = [{value:1, title:'Ityem 1'}, {value:2, title:'Ityem 2'}, {value:3, title:'Ityem 3'}];
		const current = 1;
		const expanded = 2;
		const prefix = 'prefix-';
		const cmp = renderer.create(
			<Menu list={list} current={current} expanded={expanded} prefix={prefix} onExpand={jest.fn()} onAction={jest.fn()}>
				<embed />
			</Menu>
		);

		expect(cmp.toJSON()).toMatchSnapshot();
	});
});
