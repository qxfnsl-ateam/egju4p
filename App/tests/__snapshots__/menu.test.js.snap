// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> DOM tree 1`] = `
<ul
  className="staff-menu list-group"
>
  <li
    className="staff-menu-item on list-group-item"
    onClick={[Function]}
    value={1}
  >
    Ityem 1
    <button
      className="staff-menu-exp"
      onClick={[Function]}
    />
  </li>
  <li
    className="staff-menu-item list-group-item"
    onClick={[Function]}
    value={2}
  >
    Ityem 2
    <button
      className="staff-menu-exp on"
      onClick={[Function]}
    />
    <embed />
  </li>
  <li
    className="staff-menu-item list-group-item"
    onClick={[Function]}
    value={3}
  >
    Ityem 3
    <button
      className="staff-menu-exp"
      onClick={[Function]}
    />
  </li>
</ul>
`;
